src/System.IO.Ports/src/System.IO.Ports.csproj

```diff
@@ -5,7 +5,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <ProjectGuid>{187503F4-BEF9-4369-A1B2-E3DC5D564E4E}</ProjectGuid>
     <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'netfx'">true</IsPartialFacadeAssembly>
-    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetGroup)' == 'netstandard' AND '$(TargetsWindows)' != 'true'">SR.PlatformNotSupported_IOPorts</GeneratePlatformNotSupportedAssemblyMessage>
+    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetGroup)' == 'netstandard' AND '$(TargetsWindows)' != 'true' AND '$(TargetsLinux)' != 'true'">SR.PlatformNotSupported_IOPorts</GeneratePlatformNotSupportedAssemblyMessage>
```

Comment:
You said it worked on BSD and OSX (or at least built) .. if so, no condition should be required on TargetsXXX .

Reply:
I did build PAL (e.g. cpp on OSX). This was primarily to verify I can use certain termios functions. 
I have done 0 testing so I was not comfortable enabling it in build. That will probably come in some next round.

Reply:
@kasper3 the purpose of porting part of the stuff to C was to be able to share that piece with Mono. There is no plan to do such change globally. Even if it could be done for corefx, it is not possible for coreclr. We need exception handling, for example.